src/Returns.php

```diff
@@ -0,0 +1,328 @@
+        }
+
+        $xml = Array2XML::createXML('open-return', $content);
+        $envelope = $xml->documentElement;
+        $envelope->setAttribute('xmlns',
+            'http://www.canadapost.ca/ws/openreturn-v2');
+        $payload = $xml->saveXML();
+
+        $response = $this->post(
+            "rs/{$this->customerNumber}/{$this->customerNumber}/openreturn",
+            [
+                'Accept' => 'application/vnd.cpc.openreturn-v2+xml',
+                'Content-Type' => 'application/vnd.cpc.openreturn-v2+xml',
+            ],
+            $payload,
+            $options
+        );
+        return $response;
+    }
+
+    /**
+     * Get open return template.
+     *
+     * @param string $endpoint
+     *   The endpoint of the file to retrieve from the Canada Post server.
+     * @param array $options
+     *   The options array.
+     *
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getOpenReturn($endpoint, array $options = []) {
+        $response = $this->getFile(
+            $endpoint,
+            'xml',
+            $options
+        );
+        return $response;
+    }
+
+    /**
+     * Get open return details.
+     *
+     * @param string $endpoint
+     *   The endpoint of the file to retrieve from the Canada Post server.
+     * @param array $options
+     *   The options array.
+     *
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getOpenReturnDetails($endpoint, array $options = []) {
+        $response = $this->getOpenReturn(
+            $endpoint,
+            $options
+        );
+        return $response;
+    }
+
+    /**
+     * Get next open return artifact.
+     *
+     * @param string $endpoint
+     *   The endpoint of the file to retrieve from the Canada Post server.
+     * @param array $options
+     *   The options array.
+     *
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getNextOpenReturnArtifact($endpoint, array $options = []) {
+        $response = $this->getArtifact(
+            $endpoint,
+            $options
+        );
+        return $response;
+    }
+
+    /**
+     * Get next open return artifact.
+     *
+     * @param string $from
+     *   The beginning range. YmdHs format, eg. 201808282359.
+     * @param string $to
+     *   The end range, defaults to current time. YmdHs format, eg. 201808282359.
+     * @param array $options
+     *   The options array.
+     *
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getAllOpenReturns($from, $to = '', array $options = []) {
+        if (empty($to)) {
+            $to = date('YmdHs');
+        }
+        $query_params = "from={$from}&to{$to}";
```

Comment:
**OPTIONAL CHANGE REQUEST**

Maybe you can calle the date validation functions here as well.